### DISCUSSION TITLE
Resetting Multiple Stores Example Doesn't Work with the Currying Required Elsewhere for Typescript Support

### DISCUSSION THREAD

**Discussion by TiKevin83** [pmndrs/zustand#2117] [Bug report] [answered]
Something is wrong with the documentation on how to reset all stores with respect to the currying needed for Typescript support:

https://docs.pmnd.rs/zustand/guides/how-to-reset-state

As is my existing code that has the extra currying () doesn't create the store properly using the example create() function. I was able to get around it by rewriting my code to not curry in my stores and to replicate the currying inside the custom create, which is actually probably better in my case because now I'm enforcing that my stores use persist and that they implement a reset function, but probably loses the extensibility desired by the example:

```
export const createPersistingStore = <T extends ResettableState>(
  f: StateCreator<T, [], [['zustand/persist', unknown]]>
) => {
  const store = _create<T>()(f);
  resetters.push(() => {
    store.getState().reset();
  });
  return store;
};
```

**Comment by dai-shi** [comment]
Is the question about testing?
@dbritto-dev can help??

Or, is there an issue without testing?

**Reply by TiKevin83** [reply]
There's an issue in the linked docs which have an example on how to write a custom create function which exposes a way to reset all stores. The example custom create function does not work, I believe because it conflicts with the currying in the example Typescript usage, which is here:

https://github.com/pmndrs/zustand/blob/main/docs/guides/typescript.md

So it is a bug in that reset documentation but not in the library itself.

I provided an example workaround to the broken create function, but it would only work for people using the persist middleware, so I created the issue in case someone knows how to write a better generic that passes through the currying needed to not specify the additional type params and then the docs could be updated to fix that.

**Reply by dbritto-dev** [reply]
@dai-shi I could take care of that, should be easy since we had the same stuff on testing guide

**Reply by dai-shi** [reply]
Thanks!!

**Reply by dbritto-dev** [reply, marked as answer]
@TiKevin83 @dai-shi here you go -> https://github.com/pmndrs/zustand/pull/2119 . BTW, I updated the codesandbox demos to reflect the latest changes on docs

**Reply by TiKevin83** [reply]
awesome, I can test that on Monday

**Reply by TiKevin83** [reply]
That fixed it, thanks!

**Reply by SebastianGrzymala** [reply]
Codesandbox links no longer work. I couldn't get the code from the [documentation](https://zustand.docs.pmnd.rs/guides/how-to-reset-state) to work with version 4.5.5. This is what worked for me::
```
export const create = (<T>(stateCreator: StateCreator<T>) => {
  const store = actualCreate(stateCreator);
  const initialState = store.getInitialState();
  storeResetFns.add(() => {
    store.setState(initialState, true);
  });
  return store;
}) as typeof actualCreate;
````
